dynamicio/handlers/keyed.py

```diff
@@ -0,0 +1,60 @@
+# pylint: disable=W0707
+"""KeyedResource class for reading and writing to different resources based on a key."""
+
+import os
+from typing import Dict, Optional
+
+import pandas as pd
+
+from dynamicio.base import BaseResource
+
+
+class KeyedResource(BaseResource):
```

Comment:
I'm not sure I've noticed any usage examples of `KeyedResource`? What does this work?

Reply:
I just noticed, I need to make this one immutable as well. 

KeyedResource is so you can have different keyed resources. Previously we called these 'environments'. So for most use cases it was either 'cloud' or 'local'. This _can_ be used for testing like it was previousy.

_However_, after chatting with Christos I understand that one other usecase might be to mock data in dev environments so one does not have to spin up various dbs/ services in dev. I envision to only use this resource for this usecase once testing is delegated to uhura.

Reply:
example
```
res = KeyedResource(
    keyed_resources={
        "cloud": S3ParquetResource(bucket="mybucket", path="mypath/file.parquet", allow_no_schema=True),
        "local": ParquetFileResource(path="mypath/file.parquet", allow_no_schema=True),
    },
    default_key="local"
)
res.set_key("cloud")

res.read()
```

Reply:
Fair enough 👏